components/centraldashboard/public/components/registration-page_test.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable max-len */
+import '@polymer/test-fixture/test-fixture';
+import 'jasmine-ajax';
+import {mockIronAjax, yeildForRequests} from '../ajax_test_helper';
+import {flush} from '@polymer/polymer/lib/utils/flush.js';
+
+import './dashboard-view';
+
+const FIXTURE_ID = 'registration-page-fixture';
+const MU_VIEW_SELECTOR_ID = 'test-registration-page';
+const TEMPLATE = `
+<test-fixture id="${FIXTURE_ID}">
+  <template>
+    <registration-page id="${MU_VIEW_SELECTOR_ID}"></registration-page>
+  </template>
+</test-fixture>
+`;
+const TEST_USER = 'test@kubeflow.org';
+const sleep = (t) => new Promise((res) => setTimeout(res, t));
+const yeildForAsync = () => sleep(1);
+
+describe('Registration Page', () => {
+    let registrationPage;
+    let flowcomplete;
+
+    beforeAll(() => {
+        jasmine.Ajax.install();
+        const div = document.createElement('div');
+        div.innerHTML = TEMPLATE;
+        document.body.appendChild(div);
+    });
+
+    beforeEach(() => {
+        document.getElementById(FIXTURE_ID).create();
+        registrationPage = document.getElementById(MU_VIEW_SELECTOR_ID);
+        flowcomplete = jasmine.createSpy('Flow Completion Event');
+        registrationPage.addEventListener('flowcomplete', flowcomplete);
+    });
+
+    afterEach(() => {
+        document.getElementById(FIXTURE_ID).restore();
+    });
+
+    afterAll(() => {
+        jasmine.Ajax.uninstall();
+    });
+
+    it('Should work in a Normal Flow', async () => {
+        mockIronAjax(
+            registrationPage.$.MakeNamespace,
+            {message: 'Success!'},
+        );
+
+        const $e = (selector) => registrationPage.shadowRoot.querySelector(selector);
+        flush();
+        await yeildForAsync(); // So the view can render
```

Comment:
I'm curious why these have been necessary. I found that the flush() function was all I needed to do to ensure that any rendering operations would finish before validating the DOM state. I just want to make sure there's a definitive reason why we're doing it so that tests don't become flaky at some later point.

Reply:
I realized I needed to add these as I was stepping through the tests. Seems like the update from the Ajax response to the view is not synchronous and the `flush()` call does not yield a tick for the async backlog to finish.

Reply:
I see. I think the difference here is that the `mockRequest` method I was previously using for tests returned a Promise that could be resolved in tests. Now that I take a closer look, is there a way that the two functions `mockRequest` and `mockIronAjax` could be merged into a single function. I notice they have the same signature but different behaviors and the latter doesn't return anything.